### DISCUSSION TITLE
Traceability of the secret NFT

### DISCUSSION THREAD

**Discussion by Crypte** [capsule-corp-ternoa/ternoa-proposals#41] [Ideas]
The secret seems to be ready soon and I'm looking forward to using it, but I think it could be even better. 

_**Concept : Adding traceability in the consultation of the Secret NFT**_

This could be a boolean or a list of addresses that have already had access (or both)

It is already existing on Ethereum with the "unlockable content", technically I do not know how it works, but the fact is that we can know if the content has already been seen or not. 

Example 1: A ticketing system hiding a QR code in each NFT for the access to an event, the secret has no real value once it has already been viewed (the ticket is already in the wild). 

Example 2: A gift card with a usable code inside, there is no more value once the person has already opened it (let's assume he is not so stupid as to sell a decrypted but unused content).

Example 3: Exclusive music from an artist hidden in a single NFT. We all agree that it will be much more valuable knowing that nobody has already listened to it.

**Comment by peshwar9** [comment]
Traceability of ownership changes for all NFTs is already a built-in feature. You can check it easily in the indexer.

**Comment by Leouarz** [comment]
I think that from the moment an NFT has a secret, we can consider all previous owners to have viewed the secret. 

For the ticket / giftcard, it should not be taken account of if it's a secondary sale.

For the exclusive music, you will know if someone had access before.

I think it can answer your issues.

**Reply by Crypte** [reply]
It doesn't really fix it since it completely destroys the secondary market... 

I found 3 examples in 1min, I think other people will find many more and won't be able to do it on Ternoa. 

The first owner is not necessarily interested in the secret of one of his NFT's, to consider it as open is to lose value stupidly in my opinion.

**Reply by peshwar9** [reply]
I see what you mean. We'll think about this.

**Comment by nicolapertosa** [comment]
We only need a flag that says the secret has already been consulted/opened.
